src/Core/src/Handlers/RefreshView/RefreshViewHandler.Android.cs

```diff
@@ -13,7 +13,9 @@ protected override void ConnectHandler(MauiSwipeRefreshLayout platformView)
 		{
 			base.ConnectHandler(platformView);
 			platformView.Refresh += OnSwipeRefresh;
-			platformView.SetProgressViewEndTarget(true, 50);
+
+			int progressViewEndOffset = 50 + platformView?.ProgressViewEndOffset ?? 0;
```

Comment:
The refresh indicator resting position is always positioned near the top of the refreshing content. With the changes, we always set the position 50 px based on whether or not there is a toolbar or actionbar present (ProgressViewEndOffset value).

Reply:
Hmm, a solution that incorporated changed the px count according to device dpi would be a more rigid solution to this, I think. But the 50 was just enough to clear the issue of the spinner getting stuck. This was a strange issue because I'm certain it's a fault on the Android side.

Reply:
https://issuetracker.google.com/issues/110463864

Reply:
It looks like there are a few things going on here

1) it looks like there's a bug with the SwipeRefreshLayout.  It's kind of odd because they fixed it two years ago, but it's not fixed in the latest androidx packages. If you look at the main branch they've added an else into the nestedscrollview code https://android-review.googlesource.com/c/platform/frameworks/support/+/1734993 which triggers a reset of the indicator so it hides when you scroll up. I tried a few paths to mimic what they are doing but I'm not sure a reliable way to do this.

2) I'm not sure why exactly, but, for whatever reason our RefreshView isn't clipping to bounds. If you use the SwipeRefreshLayout in a platform app you'll see the indicator goes outside the bounds of the refreshview so it hides itself out of view. My guess is that this is related to our layouts vs using androids in the box layouts. 

3) Calling any of the SetProgress* APIs works around the [original](https://github.com/dotnet/maui/pull/17080) issue because calling any of those APIs sets the `mUsingCustomStart` private variable on the SwipeRefreshLayout which causes this path, so the visibility of the spinner will get set to `Gone` once you've scrolled all the way back to zero. 

<img width="631" alt="image" src="https://github.com/dotnet/maui/assets/5375137/84432d76-9a4e-4603-a7ac-81dff2ec81a1">

The size of the ToolBar is kind of irrelevant here. Like if you just call this
`SetProgressViewOffset(false, ProgressViewStartOffset, ProgressViewEndOffset);`

It works "fine"

if you call this
`SetProgressViewOffset(false, 0, ProgressViewEndOffset);`

It also works fine and then the refresh circle doesn't go above the `RefreshView`.

I'm not quite sure where y'all got 50 from, and what that's trying to accomplish? 

If we want to somewhat replicate the values inside the `SwipeRefreshLayout` then I think we'd call it like this

`SetProgressViewOffset(true, ProgressViewStartOffset, ProgressViewEndOffset - Math.Abs(ProgressViewStartOffset));`

Which I just pulled from here

<img width="834" alt="image" src="https://github.com/dotnet/maui/assets/5375137/629a3e87-553d-49d2-bc80-47a225e95c2b">

That causes the indicator when refreshing to land in the same spot. I like setting the scale to true. I think the non scaling version is odd, especially if there's a toolbar.


Also, I would call `SetProgressViewOffset` inside the ctor of `SwipeRefreshLayout`, that way if developers want to call `SetProgressViewOffset` themselves they don't have to figure out when we call it.

Reply:
Once we merge this if we could add a manual test that'd be cool!